src/views/Moderation/Moderation.js

```diff
@@ -0,0 +1,430 @@
+import {hasAffiliatedOrganizations} from '../../utils/user'
+import {push} from 'react-router-redux'
+
+const {TABLE_DATA_SHAPE, PUBLICATION_STATUS} = constants
+
+export class Moderation extends React.Component {
+
+    state = {
+        draftData: {
+            events: [],
+            count: null,
+            paginationPage: 0,
+            pageSize: 10,
+            fetchComplete: true,
+            sortBy: 'last_modified_time',
+            sortDirection: 'desc',
+            tableColumns: ['checkbox', 'name', 'publisher', 'event_time', 'last_modified_time', 'validation'],
+            selectedRows: [],
+            invalidRows: [],
+        },
+        publishedData: {
+            events: [],
+            count: null,
+            paginationPage: 0,
+            pageSize: 10,
+            fetchComplete: true,
+            sortBy: 'date_published',
+            sortDirection: 'desc',
+            tableColumns: ['name', 'publisher', 'event_time', 'date_published'],
+        },
+    }
+
+    componentDidMount() {
+        const {user} = this.props
+
+        if (!isNull(user) && hasAffiliatedOrganizations(user)) {
+            this.fetchTableData(['draft', 'published'])
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const {user, routerPush} = this.props
+        const oldUser = prevProps.user
+
+        // redirect to home if user logged out
+        if (isNull(user) && oldUser !== user) {
+            routerPush('/')
+        }
+        // fetch data if user logged in
+        if (isNull(oldUser) && user && hasAffiliatedOrganizations(user)) {
+            this.fetchTableData(['draft', 'published'])
+        }
+    }
+
+    /**
+     * Fetches the data for the given table(s) and saves it to the state
+     * @param tables    The table(s) that data should be fetched for
+     */
+    fetchTableData = (tables) => {
+        // promises containing the table data
+        const fetchedData = tables.map(table => {
+            const queryParams = this.getDefaultEventQueryParams(table)
+            queryParams.include = table === 'draft' ? 'keywords,sub_events' : null
+            queryParams.admin_user = table === 'draft' ? true : null
+
+            return fetchEvents(queryParams)
+        })
+
+        this.setLoading(false, tables)
+
+        Promise.all(fetchedData)
+            .then(values => {
+                const tableKeys = tables.map(table => `${table}Data`)
+                // map fetched data to the tables
+                const tableData = zipObject(tableKeys, values)
+                let updatedState = {...this.state}
+
+                each(tableData, (values, key) => {
+                    updatedState[key] = {
+                        ...this.state[key],
+                        events: values.data.data,
+                        count: values.data.meta.count,
+                        selectedRows: [],
+                    }
+                })
+
+                this.setState({...updatedState})
```

Comment:
When referencing the current state, it's advised to pass an updater function to `this.setState(state => ...)` to make sure the referenced state is up-to-date.